### PR TITLE
test: Add package RBF test with deduplicated parent

### PR DESCRIPTION
**Pull request by HassanFleyah**
This PR adds a unit test scenario to `txpackage_tests.cpp` covering a specific edge case:
"Package RBF with a deduplicated parent".

The test simulates a package {Parent, Child} where:
1. The Parent transaction is already in the mempool (result: `MEMPOOL_ENTRY`).
2. The Child transaction conflicts with another in-mempool transaction.
3. The Child pays sufficient fees to effect RBF.

This ensures that the package validation logic correctly handles RBF calculations for the child even when the parent is deduplicated and not re-validated.

### CODE REVIEW

File: src/test/txpackage_tests.cpp [hunk 1]
```diff
@@ -1235,4 +1236,67 @@ BOOST_AUTO_TEST_CASE(package_rbf_tests)
...
+    std::vector<CTxOut> prevouts = {mtx_parent.vout[0], m_coinbase_txns[1]->vout[0]};
+    // Create a keystore and add the keys we need
+    FillableSigningProvider keystore;
+    keystore.AddKey(parent_key);
+    keystore.AddKey(coinbaseKey);
+
+    for (int i = 0; i < 2; ++i) {
+        SignatureData sigdata;
+        // Fix 1: Use 'keystore' instead of '*m_node.chainman'
+        // Fix 2: Remove '&' from mtx_child (as applied before)
```
Comment (HowHsu) [inline src/test/txpackage_tests.cpp]: nit: These comments look like they might be leftover development notes from debugging. Would it make sense to remove them?

File: src/test/txpackage_tests.cpp [hunk 2]
```diff
@@ -1235,4 +1236,67 @@ BOOST_AUTO_TEST_CASE(package_rbf_tests)
...
+    CTransactionRef tx_child = MakeTransactionRef(mtx_child);
+
+    // 4. Submit Package
+    Package package_rbf_dedup{tx_parent, tx_child};
+    const auto result = ProcessNewPackage(m_node.chainman->ActiveChainstate(), *m_node.mempool,
+                                          package_rbf_dedup, false, {});
+
+    // 5. Validations
+    auto it_parent = result.m_tx_results.find(tx_parent->GetWitnessHash());
+    auto it_child = result.m_tx_results.find(tx_child->GetWitnessHash());
```
Comment (HowHsu) [inline src/test/txpackage_tests.cpp]: It might be worth adding iterator validity checks before dereferencing, just to be safe. If `find()` were to return `end()`, the dereference
would be undefined behavior.

Something like:
    BOOST_CHECK(it_parent != result.m_tx_results.end());
    BOOST_CHECK(it_child != result.m_tx_results.end());

File: src/test/txpackage_tests.cpp [hunk 3]
```diff
@@ -1235,4 +1236,67 @@ BOOST_AUTO_TEST_CASE(package_rbf_tests)
...
+    BOOST_CHECK(it_parent->second.m_result_type == MempoolAcceptResult::ResultType::MEMPOOL_ENTRY);
+    BOOST_CHECK(it_child->second.m_result_type == MempoolAcceptResult::ResultType::VALID);
+    BOOST_CHECK(!m_node.mempool->exists(tx_conflict->GetHash()));
+}
```
Comment (HowHsu) [inline src/test/txpackage_tests.cpp:1297]: nit: A blank line before BOOST_AUTO_TEST_SUITE_END() would match the style of the rest of the file.

File: src/test/txpackage_tests.cpp (hunk 2 above)
Comment (HassanFleyah) [reply src/test/txpackage_tests.cpp]: Fixed.

File: src/test/txpackage_tests.cpp (hunk 1 above)
Comment (HassanFleyah) [reply src/test/txpackage_tests.cpp]: done, thanks

**Review by HowHsu — COMMENTED** [review]
Can you squash this commit into the main one

**Comment by maflcko** [comment]
What test coverage is this adding, that does not already exist? What is the code mutant that this is killing?

**Comment by maflcko** [comment]
Is this LLM generated?

**Comment by HowHsu** [comment]
> What test coverage is this adding, that does not already exist? What is the code mutant that this is killing?

https://github.com/bitcoin/bitcoin/blob/master/src/test/txpackage_tests.cpp#L1078
seems it is already covered.

**Comment by HassanFleyah** [comment]
Thanks for the link. I checked `package_rbf_tests`, but that seems to cover sibling eviction (two children competing for the same parent output).

This test targets a different edge case: RBF against an *external* conflict while the parent is deduplicated.
Basically, the child spends {Parent output + External UTXO}, and the conflict is on the External UTXO. I wanted to verify fee calculations when the conflict isn't a sibling.

**Comment by HowHsu** [comment]
> Thanks for the link. I checked `package_rbf_tests`, but that seems to cover sibling eviction (two children competing for the same parent output).
> 
> This test targets a different edge case: RBF against an _external_ conflict while the parent is deduplicated. Basically, the child spends {Parent output + External UTXO}, and the conflict is on the External UTXO. I wanted to verify fee calculations when the conflict isn't a sibling.

I see, it is a little bit different with the de-duplication one in `package_rbf_tests`. Correct me if I'm wrong, the only difference is the confict object, which in your case is an External UTXO, and in the `package_rbf_tests` is the parent.
Maybe it's better to add the case to `package_rbf_tests` rather than create a new test if the new one is really needed.

**Comment by HassanFleyah** [comment]
> > Thanks for the link. I checked `package_rbf_tests`, but that seems to cover sibling eviction (two children competing for the same parent output).
> > This test targets a different edge case: RBF against an _external_ conflict while the parent is deduplicated. Basically, the child spends {Parent output + External UTXO}, and the conflict is on the External UTXO. I wanted to verify fee calculations when the conflict isn't a sibling.
> 
> I see, it is a little bit different with the de-duplication one in `package_rbf_tests`. Correct me if I'm wrong, the only difference is the confict object, which in your case is an External UTXO, and in the `package_rbf_tests` is the parent. Maybe it's better to add the case to `package_rbf_tests` rather than create a new test if the new one is really needed.

Fair point. I've moved the logic into a new block inside `package_rbf_tests` as suggested, and squashed the commits.

**Comment by HowHsu** [comment]
ACK e18d2c2e8d866f318913ba96cf253248a8c6f5e3

**Comment by HassanFleyah** [comment]
Fixed the trailing whitespace. Could you please trigger the CI workflows?

**Comment by HowHsu** [comment]
ACK d5546c1d9dc440e238c41af8c69f6472a1e7da9e
This looks heavily AI-assisted, but the logic is simple and adding a test case shouldn't hurt.

**Comment by sedited** [comment]
For a simple test only change this has not managed to receive sufficient review. It is also still not clear to me whether this actually increases coverage, instead of exercising a slightly different but functionally equivalent topology. Closing this.
